lib/customValidators.js

```diff
@@ -239,9 +196,61 @@ function isSrvrSecProperArray(schemaType, specialSecTypes, secObj, secName) {
   return true;
 }
 
+/**
+ * Validates if parameters specified in the channel have corresponding parameters object defined and if name does not contain url parameters
+ * 
+ * @private
+ * @param  {Object} parsedJSON parsed AsyncAPI document
+ * @param  {String} asyncapiYAMLorJSON AsyncAPI document in string
+ * @param  {String} initialFormat information of the document was oryginally JSON or YAML
+ * @returns {Boolean} true in case the document is valid, otherwise throws ParserError
+ */
+function validateChannels(parsedJSON, asyncapiYAMLorJSON, initialFormat) {
+  const chnls = parsedJSON.channels;
+  if (!chnls) return true;
+
+  const chnlsMap = new Map(Object.entries(chnls));
+  const notProvidedParams = new Map(); //return object for missing parameters
+  const invalidChannelName = new Map(); //return object for invalid channel names with query parameters
+
+  chnlsMap.forEach((val, key) => {
+    const variables = parseUrlVariables(key);
+    const notProvidedChannelParams = notProvidedParams.get(tilde(key));
+    const queryParameters = parseUrlQueryParameters(key);
+    
+    //channel variable validation: fill return obeject with missing parameters
+    if (variables) {
+      setNotProvidedParams(variables, val, key, notProvidedChannelParams, notProvidedParams);
+    }
+
+    //channel name validation: fill return object with channels containing query parameters
+    if (queryParameters) {
+      invalidChannelName.set(tilde(key),
+        queryParameters);
+    }
+  });
+
+  //combine validation errors of both checks and output them as one array
+  const parameterValidationErrors = groupValidationErrors('channels', 'channel does not have a corresponding parameter object for', notProvidedParams, asyncapiYAMLorJSON, initialFormat);
+  const nameValidationErrors = groupValidationErrors('channels', 'channel contains invalid name with url query parameters ', invalidChannelName, asyncapiYAMLorJSON, initialFormat);
```

Comment:
```suggestion
  const nameValidationErrors = groupValidationErrors('channels', 'channel contains invalid name with url query parameters', invalidChannelName, asyncapiYAMLorJSON, initialFormat);
```

Reply:
all is good ready to merge, just please remove this empty space as later in the error it shows with empty space before colon `channel contains invalid name with url query parameters : ` which looks strange and is not consistent with other errors